### DISCUSSION TITLE
Provide some example of using a custom parser in a grammar  ?

### DISCUSSION THREAD

**Discussion by sandwoodK** [foonathan/lexy#194] [General]
Please apologies / feel free to correct if terminology is not good.

I've read (but can't find it anymore) that Lexy has the ability to switch to a completely hand written rule.

Let's assume that at some point I need to call a (provided) function with following signature
  `parseToSomeStuff(InputIterator & it, const InputIterator & end)`
  
 this function update the it to the last consumed value. 
 
The issue is that  end Iterator is not provided. 

What would you recommand ?

**Comment by sandwoodK** [comment]
No answer to this one ? is the question somehow unclear ?

**Reply by foonathan** [reply]
Sorry, I thought I replied already. It is currently not supported. You can use `dsl::scan` and get the current position, but you cannot advance it. I'll add something when I have time, but I'm really busy at the moment.
